lib/rules/base.js

```diff
@@ -1,163 +1,208 @@
-        'objects': 'never',
-        'imports': 'never',
-        'exports': 'never',
-        'functions': 'never'
-      }
+      'never'
     ],
+    // enforces a style where it enforces no space before comma but requires a space after comma
     'comma-spacing': [
       'error',
       {
         'before': false,
         'after': true
       }
     ],
+    // enforces a style where it enforces commas to be placed at the end of the current line
     'comma-style': [
       'error',
       'last'
     ],
+    // enforces a style where constructors of derived classes must call super()
     'constructor-super': 'error',
+    // enforces a style to never omit curly braces around blocks
     'curly': [
-      'error',
-      'multi-line'
+      'error'
     ],
+    // enforces a style to require the dot to be on the same line as the property
     'dot-location': [
       'error',
       'property'
     ],
+    // enforces a style to require at least one newline at the end of non-empty files.
     'eol-last': 'error',
+    // enforces a style to require to use type-safe equality operators === and !== instead of their regular counterparts == and !=
     'eqeqeq': [
       'error',
       'always',
       {
         'null': 'ignore'
       }
     ],
+    // enforces a style to disallow space between the function name and the opening parenthesis.
     'func-call-spacing': [
       'error',
       'never'
     ],
+    // enforces a style to enforce spacing around the * of generator functions.
     'generator-star-spacing': [
       'error',
       {
         'before': true,
         'after': true
       }
     ],
+    // enforces a style to enforce handling of error, if using the callback pattern in Node.js
     'handle-callback-err': [
       'error',
       '^(err|error)$'
     ],
+    // enforces a style to enforce 2-space indentation and 1-space for case clauses in switch statements
     'indent': [
       'error',
       2,
       {
         'SwitchCase': 1
       }
     ],
+    // enforces a style to require disallow space before colon in object literal properties but require after
     'key-spacing': [
       'error',
       {
         'beforeColon': false,
         'afterColon': true
       }
     ],
+    // enforces a style to require spaces around JavaScript Keywords
     'keyword-spacing': [
       'error',
       {
         'before': true,
         'after': true
       }
     ],
+    // enforces a style to require
+    //   all new operators to be called with uppercase-started functions.
+    //   all uppercase-started functions to be called with new operators.
     'new-cap': [
       'error',
       {
         'newIsCap': true,
         'capIsNew': false
       }
     ],
+    // enforces a style to require parentheses when invoking a constructor with no arguments using the new keyword
     'new-parens': 'error',
+    // enforces a style to disallow Array constructors. [DISCUSS]
     'no-array-constructor': 'error',
```

Comment:
IMO I would disable it as creating an array by setting the size first is not very practical in dynamic language like JS